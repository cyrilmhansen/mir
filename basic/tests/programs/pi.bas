10 D=0
20 N=1
30 WHILE 1+1/N<>1
40 N=N*10
50 D=D+1
60 WEND
70 D=D-2
80 N=1
90 FOR I=1 TO D:N=N*10:NEXT I
100 T0=TIMER
110 A1=1/5
120 A1SQ=A1*A1
130 S1=0
140 T=A1
150 I=1
160 SIGN=1
170 WHILE ABS(T)>1/N
180 S1=S1+SIGN*T/I
190 T=T*A1SQ
200 I=I+2
210 SIGN=-SIGN
220 WEND
230 A2=1/239
240 A2SQ=A2*A2
250 S2=0
260 T=A2
270 I=1
280 SIGN=1
290 WHILE ABS(T)>1/N
300 S2=S2+SIGN*T/I
310 T=T*A2SQ
320 I=I+2
330 SIGN=-SIGN
340 WEND
350 PI=16*S1-4*S2
360 T1=TIMER
370 PRINT "PI=";INT(PI);".";
380 F=PI-INT(PI)
390 FOR I=1 TO D
400 F=F*10
410 PRINT INT(F);
420 F=F-INT(F)
430 NEXT I
440 PRINT
450 PRINT "TIME=";INT(T1-T0)
460 END
