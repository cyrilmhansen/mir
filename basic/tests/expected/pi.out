PI=3.14159265358979
TIME=0
